modules/common/security/fleet/default.nix

```diff
@@ -0,0 +1,228 @@
+          ORBIT_FLEET_MANAGED_HOST_IDENTITY_CERTIFICATE = nullOrBoolToString cfg.fleetManagedHostIdentityCertificate;
+          ORBIT_HOSTNAME_FILE = cfg.hostnameFile;
+          ORBIT_INSECURE = nullOrBoolToString cfg.insecure;
+
+          # Hardcoded variables to ensure we play nice with nix.
+          ORBIT_DISABLE_KEYSTORE = "true";
+          ORBIT_DISABLE_UPDATES = "true";
+          ORBIT_FLEET_DESKTOP = "false";
+          ORBIT_LOG_FILE = "/var/log/orbit/orbit.log";
+          ORBIT_OSQUERY_DB = "${cfg.rootDir}/osquery.db";
+          ORBIT_ROOT_DIR = cfg.rootDir;
+          NIX_ORBIT_OSQUERYD_PATH = "${cfg.osqueryPackage}/bin/osqueryd";
+          NIX_ORBIT_OSQUERY_LOG_PATH = "/var/log/orbit/osquery/";
+        }
+        (lib.optionalAttrs (cfg.hostnameFile != null) {
+          OSQUERY_HOSTNAME_FILE = cfg.hostnameFile;
+          OSQUERY_UUID_FILE = "/etc/common/ghaf/uuid";
+        })
+        (lib.optionalAttrs (cfg.hostIdentifier != null) {
+          ORBIT_HOST_IDENTIFIER = cfg.hostIdentifier;
+        })
+      ];
+
+      preStart = lib.mkIf (cfg.hostIdentifier == "specified") ''
+        uuid_file="/etc/common/ghaf/uuid"
+        flagfile="${cfg.rootDir}/osquery.flags"
+        if [ -r "$uuid_file" ]; then
+          uuid="$(${pkgs.coreutils}/bin/tr -d '\n' < "$uuid_file")"
+          if [ -n "$uuid" ]; then
+            ${pkgs.coreutils}/bin/mkdir -p "${cfg.rootDir}"
+            ${pkgs.coreutils}/bin/printf -- "--specified_identifier=%s\n" "$uuid" > "$flagfile"
+          else
+            echo "orbit: uuid file empty: $uuid_file" >&2
+          fi
+        else
+          echo "orbit: uuid file not readable: $uuid_file" >&2
+        fi
+      '';
+
+      serviceConfig = {
+        ExecStart = "${cfg.orbitPackage}/bin/orbit";
+        TimeoutStartSec = 0;
+        Restart = "always";
+        RestartSec = 60;
+        KillMode = "control-group";
+        KillSignal = "SIGTERM";
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d ${cfg.rootDir} 0755 root root -"
+      "d /var/log/orbit 0755 root root -"
+    ];
+
+    # This service replaces the built in orbit logic for detecting running
+    # sessions and starting the fleet-desktop.
+    systemd.user.services."fleet-desktop" = {
+      description = "Fleet Desktop GUI";
+      after = [
+        "graphical-session.target"
+        "orbit.service"
+      ];
+      wantedBy = [ "graphical-session.target" ];
+
+      environment = {
+        FLEET_DESKTOP_DEVICE_IDENTIFIER_PATH = "${cfg.rootDir}/identifier";
+        FLEET_DESKTOP_FLEET_URL = cfg.fleetUrl;
+
+        # NOTE: We override the serviceConfig.Environment path to ensure we
+        # have xdg-utils, and all the user paths, otherwise opening the browser
+        # can fail on some systems (depending on the default they've set).
+        PATH = lib.mkForce "${pkgs.xdg-utils}/bin:/home/%u/.local/bin:/home/%u/.nix-profile/bin:/etc/profiles/per-user/%u/bin:/run/current-system/sw/bin:/usr/bin:/bin";
```

Comment:
are all the user `/home/*` paths needed? also what can the user do with this fleet desktop solution?

the desktop has the statement `Self-remediation is only available for users with Fleet Premium`. so the only benifit of the desktop is a status view?